module_utils/juniper_junos_common.py

```diff
@@ -652,7 +669,7 @@ def __init__(self,
         # Update argument_spec with the top_spec
         argument_spec.update(top_spec)
         # Extend mutually_exclusive with connection_mutually_exclusive
-        mutually_exclusive += top_spec_mutually_exclusive
+        mutually_exclusive = top_spec_mutually_exclusive
```

Comment:
This will break things! A module may specify parameters which are mutually exclusive. The common parameters which are mutually exclusive should be **added** to the module specified parameters, not override them.